src/instructlab/cli/data/list.py

```diff
@@ -17,6 +20,32 @@
 logger = logging.getLogger(__name__)
 
 
+def extract_model_and_timestamp(
+    filename: str,
+) -> Tuple[Optional[str], bool, Optional[str]]:
+    """
+    Extracts the model name and complete timestamp from a filename.
+    """
+    # Extract model name and full timestamp
+    pattern_model = r"^(test|messages|train)_(?P<model_name>[\w\-\.]+)_(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}_\d{2}_\d{2})"
```

Comment:
nit: Might be good to pull the timestamp regex out as a constant that can be shared

Reply:
Done